src/common/substrait/src/df_logical.rs

```diff
@@ -412,6 +476,24 @@ impl DFLogicalSubstraitConvertor {
     }
 }
 
+fn same_schema_without_metadata(lhs: &ArrowSchemaRef, rhs: &ArrowSchemaRef) -> bool {
+    lhs.fields.len() == rhs.fields.len()
+        && lhs.fields.iter().zip(rhs.fields.iter()).all(|(x, y)| {
+            x.name == y.name && x.data_type == y.data_type && x.is_nullable == y.is_nullable
+        })
+}
```

Comment:
Looks like it has implemented `Eq` trait https://docs.rs/arrow2/0.10.1/arrow2/datatypes/struct.Schema.html#impl-Eq